01_RecordSelectors/popup/popup.ts

```
@@ -0,0 +1,571 @@
+            if (apiKey) {
+                saveApiKeyBtn.disabled = true;
+                saveApiKeyBtn.textContent = 'Saving...';
+                chrome.runtime.sendMessage({ action: 'saveApiKey', apiKey: apiKey }, (response?: { success: boolean; error?: string }) => {
+                     if (saveApiKeyBtn) { // Check if button still exists
+                         saveApiKeyBtn.disabled = false;
+                         saveApiKeyBtn.textContent = 'Save Key';
+                     }
+                    if (chrome.runtime.lastError) {
+                        console.error("Error saving API key:", chrome.runtime.lastError.message);
+                        displayStatusMessage(`Error saving key: ${chrome.runtime.lastError.message}`, 'error');
+                    } else if (response && response.success) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)